src/IW/Sync/Search.hs

```diff
@@ -1,97 +1,139 @@
+-- | Search all repositories built with the Haskell language by page within a date range.
+searchHaskellReposByDate
+    :: Day -- ^ First day of date range that repos were created in
+    -> Day -- ^ Last day of date range that repos were created in
+    -> Int -- ^ Number of page to be returned
+    -> IO (Either GitHub.Error (SearchResult GitHub.Repo))
+searchHaskellReposByDate = githubSearch ["search", "repositories"] "language:haskell"
 
--- | Construct a github search query from a list of labels.
-labelsToSearchQuery :: [Label] -> Text
-labelsToSearchQuery = foldMap (\Label{..} -> "label:\"" <> unLabel <> "\" ")
+-- | Search all open issues with Haskell language.
+searchAllHaskellIssues
+    :: Day -- ^ First day of date range that issues were created in
+    -> Day -- ^ Last day of date range that issues were created in
+    -> Int -- ^ Number of page to be returned
+    -> IO (Either GitHub.Error (SearchResult GitHub.Issue))
+searchAllHaskellIssues = searchHaskellIssuesByLabels []
 
--- | Search for all open Haskell issues with the corresponding labels.
-searchHaskellIssuesByLabels :: [Label] -> IO (Either GitHub.Error (SearchResult GitHub.Issue))
-searchHaskellIssuesByLabels labels = searchIssues $ "language:haskell is:open " <> labelsToSearchQuery labels
+-- | Search all open issues with Haskell language and the labels passed in to the function.
+searchHaskellIssuesByLabels
+    :: [Label] -- ^ Issue labels
+    -> Day     -- ^ First day of date range that issues were created in
+    -> Day     -- ^ Last day of date range that issues were created in
+    -> Int     -- ^ Number of page to be returned
+    -> IO (Either GitHub.Error (SearchResult GitHub.Issue))
+searchHaskellIssuesByLabels labels = githubSearch ["search", "issues"] queryString
+  where
+    queryString :: Text
+    queryString = "language:haskell " <> labelsToSearchQuery labels
+
+    -- | Construct a github search query from a list of labels.
+    labelsToSearchQuery :: [Label] -> Text
+    labelsToSearchQuery = foldMap (\Label{..} -> "label:\"" <> unLabel <> "\" ")
+
+-- | Executes a query against the GitHub Search API.
+githubSearch
+    :: FromJSON a
+    => Paths -- ^ URL paths
+    -> Text  -- ^ Query string for GitHub search
+    -> Day   -- ^ First day of date range that values were created in
+    -> Day   -- ^ Last day of date range that values were created in
+    -> Int   -- ^ Number of page to be returned
+    -> IO (Either GitHub.Error (SearchResult a))
+githubSearch paths queryString from to page = executeRequest' $ buildGithubQuery paths queryString from to page
+
+-- | Function for building a GitHub search query value.
+buildGithubQuery
+    :: Paths -- ^ URL paths
+    -> Text  -- ^ Query string for GitHub search
+    -> Day   -- ^ First day of date range that values were created in
+    -> Day   -- ^ Last day of date range that values were created in
+    -> Int   -- ^ Number of page to be returned
+    -> GitHub.GenRequest 'GitHub.MtJSON 'GitHub.RO a
```

Comment:
Heh, these are very ugly types from the `github` library. But nothing we can do here